dto/api_input.py

```diff
@@ -121,14 +123,64 @@ class InputUserLoginDTO(BaseDTO):
 class UpdateUserPreferencesDTO(BaseDTO):
     preferences = fields.List(fields.Int, required=True)
     user_id = fields.Int(required=False, missing=None)
+    type = fields.Str(required=False, missing=None)
 
 
 class GetUserPreferencesDTO(BaseDTO):
     user_id = fields.Int(required=False, missing=None)
+    type = fields.Str(required=False, missing=None)
 
 
 class InputUserByGoogleDTO(BaseDTO):
     email = fields.Str(required=True)
     id = fields.Str(required=False, missing=None)
     verified_email = fields.Bool(required=False, missing=None)
-    picture = fields.Str(required=False, missing=None)
\ No newline at end of file
+    picture = fields.Str(required=False, missing=None)
+
+
+class SportPreferenceDTO:
+    def __init__(self):
+        self.main_table = "UserPreference"
+        self.related_table = "Sport"
+        self.user_id_field = "users_id"
+        self.type_id_field = "sports_id"
+        self.related_name = "sport_name"
+        self.related_logo = "sport_img"
+        self.related_id = "sport_id"
+
+
+class TeamPreferenceDTO:
+    def __init__(self):
+        self.main_table = "UserClubPreferences"
+        self.related_table = "TeamIndex"
+        self.user_id_field = "users_id"
+        self.type_id_field = "preferences"
+        self.related_name = "name"
+        self.related_logo = "logo"
+        self.related_id = "team_index_id"
+
+
+class TablesAndColumnsDTO:
```

Comment:
Firstly - it's only used for prefs and cannot possible be used for anything else, therefore your naming is incorrect.

Secondly, this is a very tricky approach which may backfire some time in the future. I am not gonna make you redesign this but I suggest you return to this task some time in the future and try to solve it again in a better way. Maybe after this project is done and presented.